lib/cachext/breaker.rb

```diff
@@ -0,0 +1,90 @@
+require "thread"
+
+module Cachext
+  class Breaker
+    attr_reader :config
+
+    def initialize(config)
+      @config = config
+    end
+
+    def for(key)
+      WithKey.new(config, key)
+    end
+
+    class WithKey
+      attr_reader :config, :key
+
+      def initialize(config, key)
+        @config = config
+        @key = key
+      end
+
+      def increment_failure
+        redis.pipelined do
+          redis.set key_str(:last_failure), Time.now.to_f
+          redis.incr key_str(:monitor)
+        end
+      end
+
+      def check_health
+        if half_open?
+          if health_check >= config.failure_threshold
```

Comment:
What actually performs this health check? It just looks like a counter right now, so maybe I'm fuzzy on the mechanism:

"half open" means we previously opened the circuit but enough time has elapsed to test to see if it's safe to let calls go through. But it looks like this code increments another counter (as requests come in?) until we exceed a failure threshold. Is the health test more than that somewhere I'm not seeing?